test/unit/local/steps/await_write_finish.js

```diff
@@ -0,0 +1,141 @@
+/* eslint-env mocha */
+/* @flow */
+
+const should = require('should')
+const awaitWriteFinish = require('../../../../core/local/steps/await_write_finish')
+const Buffer = require('../../../../core/local/steps/buffer')
+
+const lastEventToCheckEmptyness = {
+  action: 'initial-scan-done',
+  kind: 'unknown',
+  path: ''
+}
+
+async function heuristicIsEmpty (buffer) {
+  const expected = await buffer.pop()
+  return (
+    (expected.length === 1 &&
+      Object.keys(expected[0]).reduce(
+        (acc, prop) =>
+          acc && expected[0][prop] === lastEventToCheckEmptyness[prop],
+        true
+      )) ||
+    console.log(expected)
+  )
+}
```

Comment:
Is `reduce` + `console.log` the best tool for this? Wouldn't *should* assertions give more useful output in case of a failure? E.g.:

```js
const initialScanDone = {
  action: 'initial-scan-done',
  kind: 'unknown',
  path: ''
}
...
should(await buffer.pop()).deepEqual([initialScanDone])
```